flytepropeller/pkg/controller/nodes/common/utils.go

```diff
@@ -79,6 +81,24 @@ func GetTargetEntity(ctx context.Context, nCtx interfaces.NodeExecutionContext)
 	return targetEntity
 }
 
+// ReadLargeLiteral reads the offloaded large literal needed by array node task
+func ReadLargeLiteral(ctx context.Context, datastore *storage.DataStore,
+	tobeRead *idlcore.Literal) error {
+	if tobeRead.GetOffloadedMetadata() == nil {
+		return fmt.Errorf("unsupported type for reading offloaded literal")
+	}
+	// read the offloaded literal
+	dataReference := tobeRead.GetOffloadedMetadata().GetUri()
```

Comment:
nit: worth checking that `tobeRead.GetOffloadedMetadata().GetUri()` is non-empty?